lib/charms/prometheus_k8s/v0/prometheus_remote_write.py

```diff
@@ -1027,32 +1052,24 @@ def alerts(self) -> dict:
             if not alert_rules:
                 continue
 
-            try:
-                scrape_metadata = json.loads(relation.data[relation.app]["scrape_metadata"])
-                identifier = ProviderTopology.from_relation_data(scrape_metadata).identifier
-                alerts[identifier] = self._transformer.apply_label_matchers(alert_rules)
-            except KeyError as e:
-                logger.warning(
-                    "Relation %s has no 'scrape_metadata': %s",
-                    relation.id,
-                    e,
-                )
```

Comment:
Could you please briefly outline why this section is dropped?

Reply:
The way that we previously built the identifier was to look at `scrape_metadata`. This did not work for rules that were being forwarded from other charms as the scrape metadata was for the agent. Given that our libraries fill in the topology labels automatically, It is okay to expect such labels to be there and simply use those to create the identifier.